runtime/parachains/src/scheduler/migration.rs

```diff
@@ -58,11 +43,35 @@ mod v0 {
 
 /// V1: Group index is dropped from the core assignment, it's explicitly computed during
 /// candidates processing.
-mod v1 {
+pub mod v1 {
 	use super::*;
+	use crate::scheduler::{self, Config, Pallet, Scheduled};
+	use frame_support::{pallet_prelude::*, traits::OnRuntimeUpgrade, weights::Weight};
 	use sp_std::vec::Vec;
 
-	pub fn migrate<T: Config>() -> Weight {
+	pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let mut weight: Weight = Weight::zero();
+
+			if StorageVersion::get::<Pallet<T>>() < STORAGE_VERSION {
+				log::info!(target: scheduler::LOG_TARGET, "Migrating scheduler storage to v1");
+				weight = weight
+					.saturating_add(migrate::<T>())
+					.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+				STORAGE_VERSION.put::<Pallet<T>>();
+			} else {
+				log::info!(
+					target: scheduler::LOG_TARGET,
+					"Scheduler storage up to date - no need for migration"
+				);
+			}
+
+			weight
```

Comment:
Should we account for the `StorageVersion::get` read in the `else` case? Right now the weight in that case is `zero()`.

Reply:
Could also just initialise the weight to 1 read, since it needs to be accounted for regardless of what happens in the conditional